src/main/webapp/app/overview/course-overview.component.ts

```diff
@@ -194,15 +204,100 @@ export class CourseOverviewComponent implements OnInit, OnDestroy, AfterViewInit
         await this.initAfterCourseLoad();
         this.sidebarItems = this.getSidebarItems();
         this.courseActionItems = this.getCourseActionItems();
+        this.updateVisibility(window.innerHeight);
+        this.updateMenuOffset();
+    }
+
+    /** Listen window resizement event by height */
+    @HostListener('window: resize', ['$event'])
+    onResize() {
+        this.dropdownOpen = false;
+        this.dropdownClickNumber = 0;
+        this.updateVisibility(window.innerHeight);
+        this.updateMenuOffset();
+    }
+
+    /** Listen click event whether on outside of the menu or one of the items in the menu to close the dropdown menu */
+    @HostListener('document: click', ['$event'])
+    onClickCloseDropdownMenu() {
+        if (this.dropdownOpen) {
+            this.dropdownClickNumber += 1;
+            if (this.dropdownClickNumber === 2) {
+                this.dropdownOpen = false;
+                this.dropdownClickNumber = 0;
+            }
+        }
+    }
+
+    /** Update sidebar item's hidden property based on the window height to display three-dots */
+    updateVisibility(height: number) {
+        const thresholds: number[] = [];
+        const itemHeight: number = 35;
+        let threshold = this.calculateThreshold();
+        this.anyItemHidden = false;
+        this.hiddenItems = [];
+
+        for (let i = 0; i < this.sidebarItems.length - 1; i++) {
+            thresholds.unshift(threshold);
+            threshold -= itemHeight;
+        }
+        thresholds.unshift(0);
+
+        this.sidebarItems.forEach((item, index) => {
+            item.hidden = height <= thresholds[index];
+            if (item.hidden) {
+                this.anyItemHidden = true;
+                this.hiddenItems.push(item);
+            }
+        });
+    }
+
+    /** Calculate dropdown-menu position based on the number of entries in the sidebar */
+    updateMenuOffset() {
+        const leftSidebarItems: number = this.sidebarItems.length - this.hiddenItems.length;
+        if (leftSidebarItems === 1) {
+            this.dropdownOffset = 0;
+            if (window.innerHeight > 370) {
+                this.dropdownOffset += (9 - this.hiddenItems.length) * 10;
+            }
+        } else {
+            this.dropdownOffset = leftSidebarItems * 50;
+        }
```

Comment:
Could you add some comments that explain the way the offset is computed? It could also make sense to define constants for numbers like `370` and `50` here